### PR TITLE
feat: introduce quiz schema versioning with V1/V2 split

### PR DESCRIPTION
**Pull request by codeincontext**
## Summary

Introduces quiz schema versioning by splitting V1 (multiple choice only) and V2 (discriminated union for all quiz types). This establishes the foundation for enhanced quiz functionality while maintaining full backward compatibility.

### 🔄 Schema Split Architecture
- **QuizV1**: Legacy multiple choice format (arrays of questions)
- **QuizV2**: Modern discriminated union supporting all quiz types
- **Conversion utilities**: Bridge between V1 and V2 formats
- **Version detection**: Simple logic - arrays are V1, objects with version field are V2

### 📁 Schema Organization
```
schemas/quiz/
├── quizV1.ts            # Legacy MC-only schema  
├── quizV2.ts            # Modern discriminated union schema
├── conversion/
│   ├── quizV1ToV2.ts    # V1 → V2 conversion functions
│   └── quizV1ToV2.test.ts # Comprehensive test coverage
└── index.ts             # Clean exports
```

### 🏗️ Versioning Strategy
- **Types**: Versioned (QuizV1, QuizV2, PatchQuizV1)
- **Functions**: Unversioned (exitQuizInstructions, mathsExitQuiz)  
- **Properties**: Unversioned (starterQuiz, exitQuiz)
- **Agents**: Unversioned names for clean API

### Core Conversion Functions
- `convertQuizV1ToV2()` - Transform V1 arrays to V2 discriminated union
- `detectQuizVersion()` - Simple version detection logic
- Full backward compatibility maintained

### Schema Formats
```typescript
// V1: Multiple choice only (existing)
type QuizV1 = Array<{
  question: string;
  answers: string[];
  distractors: string[];
}>;

// V2: All quiz types (new)
type QuizV2 = {
  version: "v2";
  questions: Array<QuizV2Question>;
};
```

## Migration Roadmap

1. **✅ Schema Split** - This PR: Establish V1/V2 versioning
2. **⏳ Silent Upgrade** - Next PR: Auto-convert V1→V2 when opening quizzes
3. **⏳ V2 Rendering** - Next PR: Render V2 quiz types in UI
4. **⏳ Enhanced Types** - Future: Full discriminated union quiz types

## Key Benefits
- **Non-breaking**: All existing V1 functionality preserved
- **Future-ready**: Infrastructure for enhanced quiz types
- **Clean separation**: Clear V1/V2 boundaries
- **Gradual migration**: Smooth transition path established

## Testing
- Comprehensive test coverage for conversion utilities
- Type safety with strict TypeScript validation
- Edge case handling for empty/malformed data

### CODE REVIEW

File: packages/aila/src/protocol/schemas/quiz/quizV2.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+// Individual question type exports for convenience
+export type QuizV2QuestionMultipleChoice = z.infer<
+  typeof QuizV2QuestionMultipleChoiceSchema
+>;
+export type QuizV2QuestionShortAnswer = z.infer<
+  typeof QuizV2QuestionShortAnswerSchema
+>;
+export type QuizV2QuestionMatch = z.infer<typeof QuizV2QuestionMatchSchema>;
+export type QuizV2QuestionOrder = z.infer<typeof QuizV2QuestionOrderSchema>;
+export type QuizV2QuestionExplanatoryText = z.infer<
```
Comment (mantagen) [inline packages/aila/src/protocol/schemas/quiz/quizV2.ts:91]: My understanding is that we won't need this as part of our schema.

From https://www.notion.so/oaknationalacademy/Quizzes-c8a40e8c43a04b0b861522df28b2accd:

> Explanatory text is used to add context which is relevant to a series of upcoming questions. 
> - An explanatory text question is not counted as one of the 6 questions in a quiz.
